### PR TITLE
make doc: switch to python 3 

### PR DESCRIPTION
**Pull request by averinevg**
- Using python 3 instead of python 2 for documentation assembling since [python 2 is EOL](https://devguide.python.org/devcycle/#end-of-life-branches)
- Unmaintained [recommonmark](https://recommonmark.readthedocs.io/en/latest/) was replaced by [myst-parser](https://myst-parser.readthedocs.io/en/latest/index.html) according to https://github.com/readthedocs/recommonmark/issues/221

### CODE REVIEW

**Comment by atamazov** [comment]
> It looks like the documentation assembling is broken, I've got the following exception after make doc...

I am afraid that this is not so. `make doc` is tested on our CI (see the `Fp32 OpenCL Debug + Codecov` stage in Jenkinsfile) and there is none exceptions. Perhaps you don't have all the necessary dependencies installed (or their versions are wrong)?

**Comment by averinevg** [comment]
@atamazov This exception is definitely caused by wrong python version. The code does not work on python 3. But [Python 2 is EOL](https://devguide.python.org/devcycle/#end-of-life-branches). Wouldn't it be better to redo it using python 3?

**Comment by averinevg** [comment]
@atamazov This exception is definitely caused by wrong python version. The code does not work on python 3. But [Python 2 is EOL](https://devguide.python.org/devcycle/#end-of-life-branches). Wouldn't it be better to redo it using python 3?

**Comment by atamazov** [comment]
> @atamazov This exception is definitely caused by wrong python version. The code does not work on python 3. But [Python 2 is EOL](https://devguide.python.org/devcycle/#end-of-life-branches). Wouldn't it be better to redo it using python 3?

@pfultz2 Paul, what is your opinion?

**Comment by pfultz2** [comment]
Its probably good to switch to python 3, but we need to update the dockerfile to install the doc dependencies with python 3 rather than python 2(i think just changing it from `pip install` to `pip3 install` in the dockerfile should work).

**Comment by atamazov** [comment]
@pfultz2 Thank you.

**Comment by atamazov** [comment]
@averinevg @pfultz2 is busy with MIGraphX so please remind him to review (and expect delays anyway). Recommendation: check the history of modified files and assign appropriate reviewers. In case of problems, please ask @junliume or @JehandadKhan to help with this.

File: doc/requirements.txt [hunk 1]
```diff
@@ -23,8 +23,8 @@
...
 ################################################################################
-sphinx==1.6.2
-breathe==4.9.1
+sphinx
+breathe
 #git+https://github.com/arximboldi/breathe@fix-node-parent
-sphinx_rtd_theme==0.4.0
-recommonmark==0.4.0
+sphinx_rtd_theme
+recommonmark
```
Comment (pfultz2) [inline doc/requirements.txt:30]: The versions should be pinned.

**Review by pfultz2 — CHANGES_REQUESTED** [review]

**Comment by averinevg** [comment]
> @averinevg @pfultz2 is busy with MIGraphX so please remind him to review (and expect delays anyway). Recommendation: check the history of modified files and assign appropriate reviewers. In case of problems, please ask @junliume or @JehandadKhan to help with this.

@atamazov Thanks, I will keep that in mind

File: doc/requirements.txt (hunk 1 above)
Comment (averinevg) [reply doc/requirements.txt]: Fixed, thanks!

**Comment by atamazov** [comment]
@averinevg Click `[Ready for review]` if this is so.

**Comment by averinevg** [comment]
> @averinevg Click `[Ready for review]` if this is so.

@atamazov Not yet. Documentation is generated with wrong formatting and broken links.

**Comment by averinevg** [comment]
Unmaintained [recommonmark](https://recommonmark.readthedocs.io/en/latest/) was replaced by [myst-parser](https://myst-parser.readthedocs.io/en/latest/index.html) according to https://github.com/readthedocs/recommonmark/issues/221

**Comment by atamazov** [comment]
Please add https://github.com/ROCmSoftwarePlatform/MIOpen/pull/1051#issuecomment-889863294 to the PR description (the topmost comment)

**Comment by atamazov** [comment]
Please open your next PR in this repo. PRs from forks:
- Require manual intervention to launch CI jobs
- Cannot be updated by other team members (if necessary, which is not very rare case) unless you grant write access to your repo

**Comment by junliume** [comment]
Ping @pfultz2 and @JehandadKhan  for review :)

**Review by JehandadKhan — APPROVED** [review]

**Pull request merged**